processing/src/main/java/org/apache/druid/query/metadata/SegmentMetadataQueryQueryToolChest.java

```diff
@@ -194,6 +194,12 @@ public boolean isCacheable(SegmentMetadataQuery query, boolean willMergeRunners)
                          .array();
       }
 
+      @Override
+      public byte[] computeResultLevelCacheKey(SegmentMetadataQuery query)
+      {
+        return computeCacheKey(query);
```

Comment:
This also needs to include "merge" and "lenientAggregatorMerge". They're not part of the segment-level cache key, but they should be part of the result-level key.

Reply:
ok, added those to result level cache key.